lti_consumer/models.py

```diff
@@ -0,0 +1,143 @@
+"""
+LTI configuration and linking models.
+"""
+from django.db import models
+
+# LTI 1.1
+from lti_consumer.lti_1p1.consumer import LtiConsumer1p1
+# LTI 1.3
+from lti_consumer.lti_1p3.consumer import LtiConsumer1p3
+from lti_consumer.utils import get_lms_base
+
+
+class LtiConfiguration(models.Model):
+    """
+    Model to store LTI Configuration for LTI 1.1 and 1.3.
+
+    This models stores references (Usage Keys) and returns LTI
+    configuration data fetching them from XBlock fields.
+
+    With the implementation of
+    https://github.com/edx/xblock-lti-consumer/blob/master/docs/decisions/0001-lti-extensions-plugin.rst
+    this model will store all LTI configuration values as a formatted JSON.
+
+    .. no_pii:
+    """
+    # LTI Version
+    LTI_1P1 = 'lti_1p1'
+    LTI_1P3 = 'lti_1p3'
+    LTI_VERSION_CHOICES = [
+        (LTI_1P1, 'LTI 1.1'),
+        (LTI_1P3, 'LTI 1.3 (with LTI Advantage Support)'),
+    ]
+    version = models.CharField(
+        max_length=10,
+        choices=LTI_VERSION_CHOICES,
+        default=LTI_1P1,
+    )
+
+    # Configuration storage
+    # Initally, this only supports the configuration
+    # stored on the block, but should be expanded to
+    # enable storing LTI configuration in this model.
+    CONFIG_ON_XBLOCK = 'CONFIG_ON_XBLOCK'
+    CONFIG_STORE_CHOICES = [
+        (CONFIG_ON_XBLOCK, 'Configuration Stored on XBlock fields'),
+    ]
+    config_store = models.CharField(
+        max_length=25,
+        choices=CONFIG_STORE_CHOICES,
+        default=CONFIG_ON_XBLOCK,
+    )
+
+    # Block location where the configuration is stored.
+    # In the future, the LTI configuration will be
+    # stored in this model in a JSON field.
+    # This field stores a UsageKey.
+    location = models.CharField(
+        max_length=255,
+        db_index=True,
+        null=True,
+        blank=True,
+    )
+
+    def _load_block(self):  # pragma: no cover
```

Comment:
How about this instead? This way, you can set the block on the config if it's passed in to `get_lti_consumer`, otherwise it'll load the block.

```python
@property
def block(self):
    block = getattr(self, '_block', None)
    if block is None:
        from xmodule.modulestore.django import modulestore
        block = self._block = modulestore().get_item(self.location)
    return block

@block.setter
def block(self, block):
    self._block = block
```